pubsub/azure/servicebus/servicebus.go

```diff
@@ -463,6 +463,12 @@ func (a *azureServiceBus) Subscribe(req pubsub.SubscribeRequest, handler pubsub.
 			sub.close(ctx)
 			cancel()
 
+			// If context was canceled, do not attempt to reconnect
+			if a.ctx.Err() != nil {
```

Comment:
I think it may be worthwhile to check that this specifically is `context.Canceled` and not DeadlineExceeded or other error contexts.